src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfilesController.php

```diff
@@ -178,6 +179,24 @@ public function bulkDeleteAction(Request $request)
         return $this->redirectToRoute('admin_profiles_index');
     }
 
+    /**
+     * Get tabs which are accessible for given profile.
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getAccessibleTabsAction(Request $request)
```

Comment:
> This route is used in Employee add/edit page ?

yes, it's used to refresh accessible tabs dropdown when employee profile is changed in the form.



> By proper rights, I mean: he has the ability to READ employee pages.

@matks, If I check Employee READ rights here, does it make sense to keep this method in `ProfilesController`? IMO I should move this method to `EmployeeController` in this case.

Reply:
> @matks, If I check Employee READ rights here, does it make sense to keep this method in `ProfilesController`? IMO I should move this method to `EmployeeController` in this case.

Good idea indeed 👍 
Can you check what happens if someone, probably by mistake, gives an employee the READ, CREATE and UPDATE rights for Profiles, but not for Employees ? I think the ajax route will return a 302. We just need to make sure it does not break the page behavior, instead the dropdown should be empty (bonus: add an error message but I dont want to spend too much time on an exotic usecase).

Reply:
I've moved this action to `EmployeeController`. 

If you don't have READ rights for Employees now you will get a 500 response, but that's because of issue with legacy layout. You will not be able to access it if you can READ (or more) Profiles, but not Employees.

Reply:
> If you don't have READ rights for Employees now you will get a 500 response, but that's because of issue with legacy layout

o_O we need to fix this (let's open an issue)